doc/interop-test-descriptions.md

```diff
@@ -679,6 +679,38 @@ Client asserts:
 by the auth library. The client can optionally check the username matches the
 email address in the key file.
 
+### google_default_credentials
```

Comment:
btw, the concept of application default credentials (which get used to authenticate the other auth interop tests) vs google default credentials is getting pretty convoluted and probably very hard for external users to understand. Let's make sure we come up with a good documentation about why all the credential types exist and that explains the basic concepts clearly.

Reply:
Ack. Some if these oddities comes down to differences in languages I think. E.g., most languages have an "application default credentials" which is just call creds. And the user combines this with a transport creds for their channel. C++ has always had a "GoogleDefaultCredentials", encapsulating transport creds with call creds, and Java and Go recently added this API. Agreed for need for docs, but this is worth an offline chat for details.